src/transformers/models/imagebind/modeling_imagebind.py

```diff
@@ -0,0 +1,2089 @@
+    def interpolate_pos_encoding(self, embeddings: torch.Tensor, height: int, width: int) -> torch.Tensor:
+        """
+        This method allows to interpolate the pre-trained position encodings, to be able to use the model on higher resolution
+        images. This method is also adapted to support torch.jit tracing.
+
+        Adapted from:
+        - https://github.com/facebookresearch/dino/blob/de9ee3df6cf39fac952ab558447af1fa1365362a/vision_transformer.py#L174-L194, and
+        - https://github.com/facebookresearch/dinov2/blob/e1277af2ba9496fbadf7aec6eba56e8d882d1e35/dinov2/models/vision_transformer.py#L179-L211
+        """
+
+        num_patches = embeddings.shape[1] - 1
+        num_positions = self.position_embeddings.shape[1] - 1
+
+        # always interpolate when tracing to ensure the exported model works for dynamic input shapes
+        if not torch.jit.is_tracing() and num_patches == num_positions and height == width:
+            return self.position_embeddings
+
+        class_pos_embed = self.position_embeddings[:, :1]
+        patch_pos_embed = self.position_embeddings[:, 1:]
+
+        dim = embeddings.shape[-1]
+
+        new_height = height // self.patch_size
+        new_width = width // self.patch_size
+
+        sqrt_num_positions = torch_int(num_positions**0.5)
+        patch_pos_embed = patch_pos_embed.reshape(1, sqrt_num_positions, sqrt_num_positions, dim)
+        patch_pos_embed = patch_pos_embed.permute(0, 3, 1, 2)
+
+        patch_pos_embed = nn.functional.interpolate(
+            patch_pos_embed,
+            size=(new_height, new_width),
+            mode="bicubic",
+            align_corners=False,
+        )
+
+        patch_pos_embed = patch_pos_embed.permute(0, 2, 3, 1).view(1, -1, dim)
+
+        return torch.cat((class_pos_embed, patch_pos_embed), dim=1)
+
+    def image_to_video(self, pixel_values: torch.FloatTensor, time_dim: int = 2, ntimes: int = 2):
+        """
+        Maps 4-dim image tensors of shape (B, C, H, W) to 5-dim video tensors, possibly repeating the image along the
+        time dimension. For example, if `time_dim == 1`, RGB images of shape (B, C, H, W) will be transformed to
+        video of shape (B, 1, C, H, W), and then the image will be repeated along the time dimension `ntimes` to get
+        shape (B, N, C, H, W).
+        """
+        if pixel_values.ndim not in [4, 5]:
+            raise ValueError(
+                f"The input `image` tensor should be 4- or 5-dimensional but has {pixel_values.ndim} dimensions."
+            )
+
+        # Add time dimension at specified dim index
+        if pixel_values.ndim == 4:
+            pixel_values = pixel_values.unsqueeze(time_dim)
+
+        # Repeat image across the time dimension ntimes.
+        if pixel_values.shape[time_dim] == 1:
+            new_shape = [1] * len(pixel_values.shape)
+            new_shape[time_dim] = ntimes
+            pixel_values = pixel_values.repeat(new_shape)
+
+        return pixel_values
```

Comment:
if decoupling it still keeps the code transformers compliant then I have pushed a commit for it :+1:

Reply:
Hi @molbap , I think I may have to revert this change as this function converts the image inputs to videos, which is indeed needed for correct format in Conv3d. But as this conversion happens in the image processing file so we already have the image inputs as videos at the beginning of model's forward call.  This invites the following condition to be true [here](https://github.com/huggingface/transformers/blob/14f6cb56fa3f13b1765135ef074be64b346b8951/src/transformers/models/imagebind/modeling_imagebind.py#L1157). But the `reduce`clips` doesn't even need to happen for images. Even if it happens the shape of the inputs will change with `ndim` being 4. So, I think this decoupling would break a few things.

cc @EduardoPach